Directory.Build.props

```diff
@@ -20,18 +20,12 @@
     <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
     <NeutralLanguage>en-US</NeutralLanguage>
     <!--
-    TODO Remove deprecated TFMs
-    -->
-    <CheckEolTargetFramework>false</CheckEolTargetFramework>
-    <NoWarn>$(NoWarn);NETSDK1138</NoWarn>
-    <!--
-    TODO Update packages to resolve vulnerability warnings
-    -->
-    <NoWarn>$(NoWarn);NU1902;NU1903</NoWarn>
-    <!--
     TODO Fix warning from Swashbuckle.AspNetCore with dotnet pack
     -->
     <NoWarn>$(NoWarn);NU5128</NoWarn>
+    <!-- TODO Remove once .NET 9.0.0 is released -->
+    <NoWarn>$(NoWarn);NU5104</NoWarn>
+    <NuGetAuditMode>direct</NuGetAuditMode>
```

Comment:
I ran into `NuGetAuditMode` as well and did some investigation I'd like to share here, in case it helps. My conclusion is that library projects, such as Swashbuckle, should keep `<NuGetAuditMode>direct</NuGetAuditMode>` in their codebase and _not_ release a patch version each time a dependent package is patched due to vulnerabilities. (It would be nice to update to non-vulnerable versions on a major release though.)

The justification for that is because of https://github.com/dotnet/runtime/pull/107161. It's part of .NET 9, but was also packported and is included in the [.NET October 2024 Updates](https://devblogs.microsoft.com/dotnet/dotnet-and-dotnet-framework-october-2024-servicing-updates/).

What I _think_ this means, is that when an app has a framework or SDK reference, plus a reference to Swashbuckle that depends on a vulnerable version of STJ, no vulnerability warning is logged during `dotnet restore`. There's no point in updating the NuGet reference in the app, because the framework itself provides a newer version, which will be used at runtime. As a result, Swashbuckle doesn't need to release a patch each time STJ is patched, to avoid users getting warnings about vulnerable dependencies.

There is further discussion at https://github.com/dotnet/runtime/issues/105120, which states that Microsoft policy is NOT to ship new versions only to satisfy vulnerable dependencies:

![image](https://github.com/user-attachments/assets/594b404d-8bbf-40ce-abcc-51fc0407eaa2)

Reply:
I was going to re-audit the build and any upgrades _just_ for NuGet Audit warnings before merging, so this is a good reminder, but there was no intention to create "churn" by doing new releases if/when there are updates for vulnerabilities. This is just about keeping the build green but also not having to add lots of extra transient version overrides for anything somewhere in the tree with its own warning.

Reply:
Just checked, and there's 46 warnings/errors with this removed.

Reply:
Sorry, forget about this whole conversation. I hadn't had a chance to try it out yet. Turns out it does _not at all_ work as I thought. The October update _only_ helps the runtime team, which can be seen in their updates from a few days ago:

![image](https://github.com/user-attachments/assets/7a9e1608-e178-4667-b2f8-cd3505f654a8)

So they removed dependencies on packages that are already part of the .NET SDK itself (`System.Text.Json` here). However, it does not help other SDKs, (nor third-party package authors), which explains why the version of `Microsoft.Extensions.Options` (included in the Web SDK) was bumped.

I suspect this could be handled better, and in a way that third-party package authors (like Swashbuckle) would benefit from, which I'm going to propose to the NuGet team.